client/src/mvc/dataset/dataset-li-edit.js

```diff
@@ -66,8 +66,9 @@ var DatasetListItemEdit = _super.extend(
                     const Galaxy = getGalaxyInstance();
                     if (Galaxy.router) {
                         ev.preventDefault();
+                        const identifier = self.model.get("element_id") || self.model.get("id");
                         Galaxy.router.push("datasets/edit", {
-                            dataset_id: self.model.attributes.id,
+                            dataset_id: identifier,
```

Comment:
`element_id` refers to `DatasetCollectionElement` instances, which you can't mix with HDA ids (`self.model.get("id")`). Unless the model change is taken into account elsewhere you're just going to display a random dataset (if it is not private).

Reply:
Interesting. Thank you for looking into this. I assumed that all model attributes starting with `element_` are specific to that element e.g. `element_type` is `hda`. Would that not mean that the `element_id` is inconsistent? How else can we retrieve the actual hda identifier?

Reply:
Let me look into this to verify I'm not making this more complicated than necessary. In general `DatasetCollectionElement`  instances can be `hda` or `hdca`, and element id should refer to the id of a `DatasetCollectionElement`. From the dataset collection element you can get the dataset, so I would change https://github.com/guerler/galaxy/blob/4db6cc3c31c52a5ff5f3af43700a4e1fdda269b3/lib/galaxy/webapps/galaxy/controllers/dataset.py#L190 to accept element_id. But let me see if we don't already have access to the hda id at this point.

Reply:
OK, this is correct because of https://github.com/guerler/galaxy/blob/4db6cc3c31c52a5ff5f3af43700a4e1fdda269b3/client/src/mvc/collection/collection-model.js#L69 ... which is maybe not a good idea because we actually have something that is the element_id, while this is actually either a HDA id or HDCA id. Still, as a bugfix this is fine.

Reply:
Cool, thanks for double checking it.